analysis/bin/vtsAnalyser.cc

```diff
@@ -508,30 +503,46 @@ int vtsAnalyser::Test() {
     b_GenPart_eta_vec.push_back(GenPart_eta[i]);
     b_GenPart_phi_vec.push_back(GenPart_phi[i]);
     b_GenPart_mass_vec.push_back(GenPart_mass[i]);
-    cout << "b_GenPart_isGenFrom_vec size : " << b_GenPart_isGenFrom_vec.size() << " , isFrom : " << isFrom << " , isFromTop : " << isFromTop << " , isFromW : " << isFromW << endl;
     for (unsigned int j=0; j<nJet; ++j) { // Loop for all of recoJet
       TLorentzVector jet_tlv;
       jet_tlv.SetPtEtaPhiM(Jet_pt[j], Jet_eta[j], Jet_phi[j], Jet_mass[j]);
-      // gen KS inside Jet 
-      if ((jet_tlv.DeltaR(gen_tlv) < 0.5) && (GenPart_pt[i]/Jet_pt[j] > 0.15)) {
-        if (isFrom == qjMapForMC_[j]) { // For jet with the same origin as gen KS
-          b_GenPart_isGenFrom_jetMat_vec.push_back(isFrom);
-          b_GenPart_isGenFromTop_jetMat_vec.push_back(isFromTop);
-          b_GenPart_isGenFromW_jetMat_vec.push_back(isFromW);
-          b_GenPart_d_jetMat_vec.push_back(GetD(GenPart_pt[i], GenPart_eta[i], GenPart_phi[i], GenPart_mass[i], gen_tlv.X(), gen_tlv.Y(), gen_tlv.Z()));
-          b_GenPart_x_jetMat_vec.push_back(GenPart_pt[i]/Jet_pt[j]);
-          b_GenPart_dr_jetMat_vec.push_back(jet_tlv.DeltaR(gen_tlv));
-          b_GenPart_pt_jetMat_vec.push_back(GenPart_pt[i]);
-          b_GenPart_eta_jetMat_vec.push_back(GenPart_eta[i]);
-          b_GenPart_phi_jetMat_vec.push_back(GenPart_phi[i]);
-          b_GenPart_mass_jetMat_vec.push_back(GenPart_mass[i]);
-        }
-      }
-    }   
+      auto dr = jet_tlv.DeltaR(gen_tlv);
+      if (dr < drReco ) { 
+        drReco = jet_tlv.DeltaR(gen_tlv);
```

Comment:
You already have jet_tlv.DeltaR(gen_tlv), you don't need to calculate it a second time

Reply:
Ah! I'll modify